File: Ejemplos/22-HarryPotterReactive/src/main/java/dev/joseluisgs/models/Spell.java
package dev.joseluisgs.models;

import lombok.Data;

@Data
public class Spell {
    private final int id;
    private final String name;
    private final Type type;

    public enum Type {
        ATTACK,
        DEFENSE,
        HEAL,
        BUFF,
        DEBUFF,
        NONE
    }
}

File: Ejemplos/31-HarryPotter-RSockets/src/main/kotlin/models/Spell.kt
package models

import kotlinx.serialization.Serializable

@Serializable
data class Spell(
    val id: Int,
    val name: String,
    val type: Type,
) {
    enum class Type {
        ATTACK,
        DEFENSE,
        HEAL,
        BUFF,
        DEBUFF,
        NONE
    }
}